sql/core/src/main/scala/org/apache/spark/sql/execution/SortExec.scala

```diff
@@ -87,13 +95,13 @@ case class SortExec(
     }
 
     val pageSize = SparkEnv.get.memoryManager.pageSizeBytes
-    val sorter = UnsafeExternalRowSorter.create(
+    rowSorter = UnsafeExternalRowSorter.create(
```

Comment:
Although createSorter should be called just once, it is better to check if rowSorter was already set.

Reply:
Thanks, done in 631f3cb.